app/scripts/modules/cloudfoundry/instance/details/instance.details.controller.js

```diff
@@ -153,8 +154,13 @@ module.exports = angular.module('spinnaker.instance.detail.cf.controller', [
       if ($scope.$$destroyed) {
         return;
       }
-      $state.params.allowModalToStayOpen = true;
-      $state.go('^', null, {location: 'replace'});
+
+      if (!instance && $scope.instanceIdNotFound) {
+        $scope.state.loading = false;
+      } else {
+        $state.params.allowModalToStayOpen = true;
+        $state.go('^', null, {location: 'replace'});
+      }
```

Comment:
same idea as above - cleaner to just set the `loading` flag to false (or rename the method to `handleInstanceNotFound` or something, and remove the `$state.go` call